### PR TITLE
Developer Experience improvements

### PR DESCRIPTION
**Pull request by TWiStErRob**
## Proposed changes

I started using this action to get the base branch, so that I can check it out and execute code from it.
While I was doing this I noticed a few things that could be improved. I bundled them up into this PR.

## Types of changes

What types of changes does your code introduce ?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update (if none of the other choices apply)

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments

See the individual commits on what changed (and sometimes why). I recommend squashing this in the end when merging. Feel free to ask about any change, happy to explain more on details or why I did it.

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -4,7 +4,8 @@ description: Get the PR associated with the current commit.
 inputs:
   github-token:
     description: The GitHub token used to create an authenticated client.
-    required: true
+    required: false
+    default: ${{ github.token }}
```
Comment (TWiStErRob) [inline action.yml:8]: does this need testing? I tried using it in my project, and have seen this in many other projects, including [competition](https://github.com/jwalton/gh-find-current-pr/blob/e12d66bc9ecc4fdcde07b0f70a3cb68ce7e4d807/action.yml#L8) 😉

**Comment by TWiStErRob** [comment]
https://github.com/TWiStErRob/net.twisterrob.sun/pull/101/commits/425245f6f462fdebe8ef1be5b84c212e17153172 is where I adopted this.

File: action.yml (hunk 1 above)
Comment (8BitJonny) [reply action.yml:8]: yeah 😄 this looks very solid. no need for extra testing

File: README.md [hunk 2]
```diff
@@ -27,26 +27,45 @@ This action enables you to get the PR no matter which event type triggered the w
...
-    - uses: actions/checkout@v1
+    - uses: 8BitJonny/gh-get-current-pr@2.0.0
+      id: PR
+
+    - run: echo "Your PR number is ${{ steps.PR.outputs.number }} and its JSON is ${{ steps.PR.outputs.pr }}"
+```
+
+### Inputs
+See [action.yml](action.yml) for more details.
+```yml
```
Comment (8BitJonny) [inline README.md:40]: Can we keep it uniform? The first code snippet starts with 'steps', this one skips it, the "Outputs" one has it again and the "Workaround" code snipped is skipping the 'steps' line again.

I don't have too much of an opinion on whether it cleaner with or without the 'steps' line but I kinda like it to be uniform. Can you change that for all the code snippets?

**Review by 8BitJonny — CHANGES_REQUESTED** [review]
Thank you so very much! I love these changes! Only one tiny comment.
I'll merge it then as soon as possible and probably have it released under v2.1.0 shortly after so that the github token is not required anymore.

Again, thanks @TWiStErRob for all the work you put into this. Much appreciated!

File: README.md (hunk 2 above)
Comment (TWiStErRob) [reply README.md:40]: oh, yes, also wanted to fix the indentation so when copied to yaml, it's at the correct level. I'll do the changes today or tomorrow.

Comment (TWiStErRob) [reply README.md:40]: added `id: PR` to all as well

**Comment by 8BitJonny** [comment]
I'm at work right now but it looks good! I'll give it a final review over the weekend and create a release out of it

**Comment by TWiStErRob** [comment]
sure, no rush

**Review by 8BitJonny — APPROVED** [review]
Looks good 🚀

**Comment by 8BitJonny** [comment]
All right, all done and release under v2.1.0
Thanks again for your contribution

**Pull request merged**